File: 프로그래머스/2/42842. 카펫/카펫.py
# 세로 * 가로 = brown + yellow
# yellow 가로 + yellow 가로 + 4 = brown
# 리턴 값 : [yellow + 가로 + 2, yellow + 세로 + 2]
# 개수가 주어짐
# yellow 가로 길이 : x, yellow 세로 길이 : y

# xy = yellow
# (x+2)(y+2) = brown

def solution(brown, yellow):
    
    s = brown+yellow
    arr = []
    for i in range(1,s):
        if (s%(i+2) == 0 and (yellow/i+2)*(i+2) == brown+yellow):
            print(i)
            arr.append([i+2,int(yellow/i)+2])
    
    answer = min(arr,key = lambda x : x[1])
    return answer

File: 프로그래머스/2/42842. 카펫/카펫.java
import java.util.*;
import java.io.*;

//yellow 약수 구해서 2x+2y+4 = brown 나오게

class Solution {
    public int[] solution(int brown, int yellow) {
        
        List<int[]> divisor = new ArrayList<>();
        
        int[] answer = {};
        if (yellow == 1) answer = new int[]{3,3};
        
        //약수 모음
        for(int i = 1; i*i<=yellow; i++){
            if (yellow%i == 0) divisor.add(new int[]{i, yellow/i});
        }
        
        for(int[] arr : divisor){
            if ((arr[0]*2)+(arr[1]*2) + 4 == brown) answer = new int[]{arr[1]+2,arr[0]+2};
        }
        return answer;
    }
}